packages/statemanager/src/ethersStateManager.ts

```diff
@@ -137,6 +137,8 @@ export class EthersStateManager extends BaseStateManager implements StateManager
       storage = accountStorage.get(key.toString('hex'))
       if (storage !== undefined) {
         return storage
+      } else {
+        return Buffer.from([])
```

Comment:
Same here, add a test to the state manager test?

Reply:
Strange behavior here.

Without the change -- it seems clear that `undefined` is a potential return value.
But if you add `undefined` to the return types -- `async getContractStorage(address: Address, key: Buffer): Promise<Buffer | undefined> {...`,
typescript complains that it now incorrectly implements the BaseStateManager method.

Is there some under-the-hood typescript magic happening that causes this to return a buffer no matter what??

Reply:
You can't add `undefined` because that would break the interface that we're implementing.  The interface [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/25223887ed2f7ec8bfd62473f165d332223b687d/packages/statemanager/src/interface.ts#L22) requires that this function returns a `buffer`. That is interesting though that Typescript didn't complain about not covering that edge case.  I think your conditional works fine though.